tests/unit/test_req_file.py

```diff
@@ -345,6 +351,63 @@ def test_nested_constraints_file(
         assert reqs[0].name == req_name
         assert reqs[0].constraint
 
+    def test_recursive_requirements_file(
+        self, tmpdir: Path, session: PipSession
+    ) -> None:
+        req_files: list[Path] = []
+        req_file_count = 4
+        for i in range(req_file_count):
+            req_file = tmpdir / f"{i}.txt"
+            req_file.write_text(f"-r {(i+1) % req_file_count}.txt")
+            req_files.append(req_file)
+
+        # When the passed requirements file recursively references itself
+        with pytest.raises(
+            RequirementsFileParseError,
+            match=(
+                f"{path_to_string(req_files[0])} recursively references itself"
+                f" in {path_to_string(req_files[req_file_count - 1])}"
```

Comment:
`re.escape` is a better way to do this.

https://docs.python.org/3/library/re.html#re.escape

Reply:
ahh nice, thanks for the tip